// ng-navigator-share/__ivy_ngcc__/fesm2015/ng-navigator-share.js
import { __awaiter, __decorate, __metadata } from 'tslib';
import { ɵɵdefineInjectable, Injectable } from '@angular/core';

import * as ɵngcc0 from '@angular/core';
let NgNavigatorShareService = class NgNavigatorShareService {
    constructor() {
        this.webNavigator = null;
        this.webNavigator = window.navigator;
    }
    canShare() {
        return this.webNavigator !== null && this.webNavigator.share !== undefined;
    }
    share({ title, text, url }) {
        return new Promise((resolve, reject) => __awaiter(this, void 0, void 0, function* () {
            if (this.webNavigator !== null && this.webNavigator.share !== undefined) {
                if ((text === undefined || text === null) &&
                    (url === undefined || url === null)) {
                    console.warn(`text and url both can't be empty, at least provide either text or url`);
                }
                else {
                    try {
                        const isShared = yield this.webNavigator.share({
                            title: title,
                            text: text,
                            url: url,
                        });
                        resolve({
                            shared: true
                        });
                    }
                    catch (error) {
                        reject({
                            shared: false,
                            error: error
                        });
                    }
                }
            }
            else {
                reject({
                    shared: false,
                    error: `This service/api is not supported in your Browser`
                });
            }
        }));
    }
};
NgNavigatorShareService.ɵfac = function NgNavigatorShareService_Factory(t) { return new (t || NgNavigatorShareService)(); };
NgNavigatorShareService.ɵprov = ɵɵdefineInjectable({ factory: function NgNavigatorShareService_Factory() { return new NgNavigatorShareService(); }, token: NgNavigatorShareService, providedIn: "root" });
NgNavigatorShareService = __decorate([ __metadata("design:paramtypes", [])
], NgNavigatorShareService);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgNavigatorShareService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();

/*
 * Public API Surface of ng-navigator-share
 */

/**
 * Generated bundle index. Do not edit.
 */

export { NgNavigatorShareService };
