// ng-navigator-share/esm2015/lib/ng-navigator-share/ng-navigator-share.service.js
import { __awaiter, __decorate, __metadata } from "tslib";
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
let NgNavigatorShareService = class NgNavigatorShareService {
    constructor() {
        this.webNavigator = null;
        this.webNavigator = window.navigator;
    }
    canShare() {
        return this.webNavigator !== null && this.webNavigator.share !== undefined;
    }
    share({ title, text, url }) {
        return new Promise((resolve, reject) => __awaiter(this, void 0, void 0, function* () {
            if (this.webNavigator !== null && this.webNavigator.share !== undefined) {
                if ((text === undefined || text === null) &&
                    (url === undefined || url === null)) {
                    console.warn(`text and url both can't be empty, at least provide either text or url`);
                }
                else {
                    try {
                        const isShared = yield this.webNavigator.share({
                            title: title,
                            text: text,
                            url: url,
                        });
                        resolve({
                            shared: true
                        });
                    }
                    catch (error) {
                        reject({
                            shared: false,
                            error: error
                        });
                    }
                }
            }
            else {
                reject({
                    shared: false,
                    error: `This service/api is not supported in your Browser`
                });
            }
        }));
    }
};
NgNavigatorShareService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgNavigatorShareService_Factory() { return new NgNavigatorShareService(); }, token: NgNavigatorShareService, providedIn: "root" });
NgNavigatorShareService = __decorate([
    Injectable({
        providedIn: 'root'
    }),
    __metadata("design:paramtypes", [])
], NgNavigatorShareService);
export { NgNavigatorShareService };
